test/support/shared_flows/redundancy_pay_flow_test_helper.rb

```diff
@@ -0,0 +1,108 @@
+module RedundancyPayFlowTestHelper
+  def test_shared_redundancy_pay_flow_questions
+    context "question: date_of_redundancy?" do
+      setup do
+        testing_node :date_of_redundancy?
+      end
+
+      should "render question" do
+        assert_rendered_question
+      end
+
+      context "validation" do
+        should "be valid for a valid redundancy date" do
+          assert_valid_response 3.years.ago.strftime("%Y-%m-%d")
+        end
+
+        should "be invalid for an invalid redundancy date" do
+          assert_invalid_response 5.years.ago.strftime("%Y-%m-%d")
+        end
+      end
+
+      context "next node" do
+        should "have a next node of age_of_employee?" do
+          assert_next_node :age_of_employee?, for_response: 3.years.ago.strftime("%Y-%m-%d")
+        end
+      end
+    end
+
+    context "question: age_of_employee?" do
+      setup do
+        testing_node :age_of_employee?
+        add_responses date_of_redundancy?: 3.years.ago.strftime("%Y-%m-%d")
+      end
+
+      should "render question" do
+        assert_rendered_question
+      end
+
+      context "validation" do
+        should "be valid for a valid employee age" do
+          assert_valid_response "16"
+        end
+
+        should "be invalid for an invalid employee age" do
+          assert_invalid_response "15"
+        end
+      end
+
+      context "next_node" do
+        should "have a next node of years_employed?" do
+          assert_next_node :years_employed?, for_response: "16"
+        end
+      end
+    end
+
+    context "question: years_employed?" do
+      setup do
+        testing_node :years_employed?
+        add_responses date_of_redundancy?: 3.years.ago.strftime("%Y-%m-%d"),
+                      age_of_employee?: "16"
+      end
+
+      should "render question" do
+        assert_rendered_question
+      end
+
+      context "validation" do
+        should "be valid for a valid number of years employe" do
+          assert_valid_response "1"
+        end
+
+        should "be invalid for an invalid number of years employed" do
+          assert_invalid_response "2"
+        end
+      end
+
+      context "next_node" do
+        should "have a next node of done_no_statutory if number of years employed is less than 2" do
+          assert_next_node :done_no_statutory, for_response: "1"
+        end
+
+        should "have a next node of weekly_pay_before_tax? if number of years employed is greater than 2" do
+          add_responses age_of_employee?: "17"
+          assert_next_node :weekly_pay_before_tax?, for_response: "2"
+        end
+      end
+    end
+
+    context "question: weekly_pay_before_tax?" do
+      setup do
+        testing_node :weekly_pay_before_tax?
+        add_responses date_of_redundancy?: 3.years.ago.strftime("%Y-%m-%d"),
```

Comment:
This threw me a bit as it feels like it should be invalid if the date of redundancy is longer ago than the number of years worked... however it is valid in the flow as I suppose it doesn't effect the calculation. Could change it to a time within the years_employed? though I also appreciate using this value keeps consistency.

Reply:
The redundancy flows doesn't assume that the redundancy occurred recently, it could have happened a few years in the past.

In reality the time frames for `years_employed?` and `date_of_redundancy?` can never overlap, because once you've been made redundant, you are no longer employed. I think the confusion comes from my use of "X years ago" for the `date_of_redundancy?` rather than using a specific date. I used "X years ago" years for the redundancy date because it passed all of the conditionals for `first_selectable_date` and `last_selectable_date` (which changes every year) without needing to resort to time travelling (with `travel_to`) to make sure the redundancy date is always valid.

`years_employed?` refers to how many years they were employed, not _when_ they were employed. So I guess for this question, they could have been employed somewhere 5 years ago, worked there for 2 years and were made redundant 3 years ago. 

`years_employed?` is also affected by the `age_of_employee?` They must have been over 17 when they were made redundant to have been able to have even worked enough years (2 or more) to claim redundancy pay.

Reply:
Thanks for explaining :) yeah sounds good to keep.